confignode/src/main/java/org/apache/iotdb/confignode/manager/ConfigManager.java

```diff
@@ -775,6 +777,65 @@ public DataSet showRegion(GetRegionInfoListPlan getRegionsinfoReq) {
     }
   }
 
+  @Override
+  public DataSet showDataNodes(GetRegionInfoListPlan getRegionsinfoReq) {
+    TSStatus status = confirmLeader();
+    DataNodeInfosResp dataNodeInfosResp = new DataNodeInfosResp();
+    if (status.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+      List<TDataNodesInfo> dataNodesInfoList = nodeManager.getOnlineDataNodesInfoList();
+      RegionInfoListResp regionsInfoDataSet =
+          (RegionInfoListResp) partitionManager.getRegionInfoList(getRegionsinfoReq);
+      Map<Integer, Integer> dataRegionNumMap = new HashMap<>();
+      Map<Integer, Integer> schemaRegionNumMap = new HashMap<>();
+      if (regionsInfoDataSet.getRegionInfoList() != null) {
+        List<TRegionInfo> regionInfoList = regionsInfoDataSet.getRegionInfoList();
+        regionInfoList.forEach(
+            (regionInfo) -> {
+              if (dataRegionNumMap.get(regionInfo.getDataNodeId()) != null) {
+                dataRegionNumMap.put(
+                    regionInfo.getDataNodeId(),
+                    dataRegionNumMap.get(regionInfo.getDataNodeId())
+                                + regionInfo.getConsensusGroupId().getType().getValue()
+                            == TConsensusGroupType.DataRegion.getValue()
+                        ? 1
+                        : 0);
+                schemaRegionNumMap.put(
+                    regionInfo.getDataNodeId(),
+                    schemaRegionNumMap.get(regionInfo.getDataNodeId())
+                                + regionInfo.getConsensusGroupId().getType().getValue()
+                            == TConsensusGroupType.SchemaRegion.getValue()
+                        ? 1
+                        : 0);
+              } else {
+                dataRegionNumMap.put(
+                    regionInfo.getDataNodeId(),
+                    regionInfo.getConsensusGroupId().getType().getValue()
+                            == TConsensusGroupType.DataRegion.getValue()
+                        ? 1
+                        : 0);
+                schemaRegionNumMap.put(
+                    regionInfo.getDataNodeId(),
+                    regionInfo.getConsensusGroupId().getType().getValue()
+                            == TConsensusGroupType.SchemaRegion.getValue()
+                        ? 1
+                        : 0);
+              }
+            });
```

Comment:
If you use AtomicInteger before, then you can use map.computeIfAbsent to facilitate this block of code~

Reply:
fixed~